moped-editor/src/views/projects/projectView/ProjectComponentsMapView.js

```diff
@@ -0,0 +1,130 @@
+import React, { useRef } from "react";
+import ReactMapGL, { NavigationControl } from "react-map-gl";
+import { Box, Button, makeStyles } from "@material-ui/core";
+import "mapbox-gl/dist/mapbox-gl.css";
+import "react-map-gl-geocoder/dist/mapbox-gl-geocoder.css";
+
+import {
+  createSummaryMapLayers,
+  getSummaryMapInteractiveIds,
+  MAPBOX_TOKEN,
+  mapStyles,
+  renderTooltip,
+  renderFeatureCount,
+  countFeatures,
+  useHoverLayer,
+  useFeatureCollectionToFitBounds,
+} from "../../../utils/mapHelpers";
+import { EditLocation as EditLocationIcon } from "@material-ui/icons";
+
+const useStyles = makeStyles({
+  locationCountText: {
+    fontSize: "0.875rem",
+    fontWeight: 500,
+  },
+  toolTip: mapStyles.toolTipStyles,
+  navStyle: {
+    position: "absolute",
+    top: 0,
+    left: 0,
+    padding: "10px",
+  },
+  editButton: {
+    position: "absolute",
+    top: 8,
+    right: 8,
+  },
+});
+
+const ProjectComponentsMapView = ({
+  projectFeatureCollection,
+  setIsEditing,
+  editEnabled,
+}) => {
+  const classes = useStyles();
+  const mapRef = useRef();
+  const featureCount = countFeatures(projectFeatureCollection);
+
+  /**
+   * Make use of a custom hook that returns a vector tile layer hover event handler
+   * and the details to place and populate a tooltip.
+   */
+  const { handleLayerHover, featureText, hoveredCoords } = useHoverLayer();
```

Comment:
the `useHoverLayer()` hook gets the job done really well, but wanted to mention for future reference that the mapbox [`feature-state`](https://docs.mapbox.com/mapbox-gl-js/api/map/#map#setfeaturestate) getter/setter might be helpful down the road.